### PR TITLE
Update message delivery guarantee section

### LINKED ISSUE #796: Improve documentation on Message Delivery Guarantees
The section is not specific enough in referring to the default binding properties in Spring Cloud Stream.  Need to document about 

* persistent pub/sub
* The relevant binding properties that control retry logic
* The relevant binding properties for DLQ

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #796

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@markpollack Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-dataflow&pullRequestId=797)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-dataflow&pullRequestId=797) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@markpollack Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-dataflow&pullRequestId=797)!

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 1]
```diff
@@ -149,18 +149,24 @@ app.http.producer.partitionKeyExpression=payload.sensorId
...
+
+Additional messaging delivery guarantees are those provided by the underlying messaging middleware that is chosen for the application for both producing and consuming applications.  Refer to the Kafka {spring-cloud-stream-docs}#_kafka_consumer_properties[Consumer] and {spring-cloud-stream-docs}#_kafka_producer_properties[Producer] and Rabbit {spring-cloud-stream-docs}#_rabbitmq_consumer_properties[Consumer] and {spring-cloud-stream-docs}#_rabbit_producer_properties[Producer] documentation for more details.  You will find there to be extensive declarative support for all the native QOS options.
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: Last sentence can be `You will find extensive declarative support for all the native QOS options.`

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 2]
```diff
@@ -114,7 +114,7 @@ the specific documentation of each Data Flow Server for more detail.
 [[spring-cloud-dataflow-stream-app-whitelisting]]
 === Whitelisting application properties
 
-Stream applications are Spring Boot applications which are aware of many link:http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#common-application-properties[common application properties], e.g. `server.port` but also families of properties such as those with the prefix `spring.jmx` and `logging`.  When creating your own application it is desirable to whitelist properties so that the shell and the UI can display them first as primary properties when presenting options via TAB completion or in drop-down boxes.
+Stream applications are Spring Boot applications which are aware of many <spring-cloud-dataflow-global-properties>, e.g. `server.port` but also families of properties such as those with the prefix `spring.jmx` and `logging`.  When creating your own application it is desirable to whitelist properties so that the shell and the UI can display them first as primary properties when presenting options via TAB completion or in drop-down boxes.
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: Should be `<<spring-cloud-dataflow-global-properties>>` for same page anchors.

**Comment by ilayaperumalg** [comment]
LGTM, will merge along with the changes to fix the comments.

**Comment by ilayaperumalg** [comment]
Rebased and merged as 1210f4ac5d64bf3f6925a09d2b65dbd73e908da7 along with 3e56c6a28f0147135c9c195e9846937fa5cef691

**Pull request closed without merging**